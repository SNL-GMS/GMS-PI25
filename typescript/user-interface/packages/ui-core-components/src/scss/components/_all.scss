@import './charts/_all.scss';
@import './widgets/_all.scss';
@import './core-table.scss';
@import './dialog.scss';
@import './d3.scss';
@import './divider.scss';
@import './overlay.scss';
@import './resizer.scss';
@import './table-ag-grid-overrides.scss';
@import './table-overrides.scss';
@import './table-cell.scss';
@import './scroll-box.scss';
@import './tooltip.scss';
