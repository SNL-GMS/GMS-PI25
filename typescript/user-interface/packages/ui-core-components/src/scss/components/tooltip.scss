.tooltip {
  .bp4-heading {
    margin-top: 0;
    margin-left: 0;
    font-weight: bold;
    &:last-child {
      // in case there is no content below
      margin-bottom: 0;
      margin-top: 0;
    }
  }
  .tooltip-content {
    display: grid;
    width: 420px;
    margin: 0 auto;
    background-color: var(--core-grid-background, $core-grid-background);

    &:last-child {
      padding-bottom: 0;
    }

    .label-value__label {
      display: none;
    }

    .label-value__value {
      background: $core-transparent;
      font-size: 14px;
      padding: 0;
      margin: 0;
      font-family: monospace;

      ul {
        display: grid;
        grid-template-rows: auto;
        grid-column-start: 1;
        justify-self: left;
        margin-top: 0;
        padding-inline-start: 0px;
        width: 100%;

        li {
          display: grid;
          grid-template-columns: 1fr;
          justify-content: space-evenly;
          box-sizing: border-box;
          padding: 0 7px;
          margin-bottom: 0.15em;
          background-color: rgba(255, 255, 255, 0.5);

          span {
            justify-self: center;
          }

          &.overflow {
            grid-template-columns: 1fr;
            span {
              justify-self: center;
            }
          }
        }
      }
    }
  }

  &.events-conflict-tooltip {
    .tooltip-content {
      .label-value__value {
        ul li {
          //                    {phase} | on | {station} | at | {time}
          grid-template-columns: 1.5fr 0.5fr 1.25fr 0.5fr 4.1fr;
          &.overflow {
            grid-template-columns: 1fr;
          }
        }
      }
    }
  }

  &.signal-detections-conflict-tooltip {
    .tooltip-content {
      width: 210px;
      .label-value__value {
        ul li {
          span {
            justify-self: start;
          }
          &.overflow {
            grid-template-columns: 1fr;
            span {
              justify-self: center;
            }
          }
        }
      }
    }
  }
}
