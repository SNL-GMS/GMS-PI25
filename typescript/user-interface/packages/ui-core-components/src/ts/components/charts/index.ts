import { BarChart } from './bar-chart';
import { LineChartWeavess } from './line-chart-weavess';
import * as ChartTypes from './types';

export { BarChart, ChartTypes, LineChartWeavess };
export * from './pie-chart';
