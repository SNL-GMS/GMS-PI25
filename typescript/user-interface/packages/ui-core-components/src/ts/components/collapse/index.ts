export * from './collapse-button';
export * from './types';
