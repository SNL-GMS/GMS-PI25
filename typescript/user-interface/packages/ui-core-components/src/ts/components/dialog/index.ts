import * as DialogTypes from './types';

export { DialogTypes };
export { DialogItem } from './dialog-item';
export { DialogTitle } from './dialog-title';
export { ModalPrompt } from './prompt';
export { SaveOpenDialog } from './save-open-dialog';
