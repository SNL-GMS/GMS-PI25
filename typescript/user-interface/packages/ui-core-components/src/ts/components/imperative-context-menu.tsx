import { Classes, ContextMenu2, hideContextMenu } from '@blueprintjs/popover2';
import { useForceUpdate } from '@gms/ui-util';
import React from 'react';

export type ImperativeContextMenuOpenFunc<T = unknown> = (
  event: React.MouseEvent | MouseEvent | Event,
  data?: T
) => void;

export type ImperativeContextMenuGetOpenCallbackFunc<T = undefined> = (
  callback: ImperativeContextMenuOpenFunc<T>
) => void;

export type ContextMenu2Content = JSX.Element | undefined;

export type ImperativeContextMenuContentFunc<T = undefined> = (data: T) => ContextMenu2Content;

export interface ImperativeContextMenuProps<T = undefined> {
  readonly content: ImperativeContextMenuContentFunc<T> | JSX.Element;
  readonly getOpenCallback: ImperativeContextMenuGetOpenCallbackFunc<T>;
  readonly onClose?: () => void;
  readonly onOpen?: () => void;
}

/**
 * @returns a hook that returns a mutable ref to a {@link ImperativeContextMenuOpenFunc} and
 * callback for setting the {@link ImperativeContextMenuOpenFunc} that can be used for an instance of a
 * {@link ImperativeContextMenu}.
 */
export function useImperativeContextMenuCallback<
  T = undefined,
  C = ImperativeContextMenuOpenFunc<T>
>(initial: C = undefined): [C, (callback: C) => void] {
  const forceUpdate = useForceUpdate();
  const contextMenuCbRef = React.useRef<C>(initial);
  const getOpenCallback = React.useCallback((callback: C) => {
    contextMenuCbRef.current = callback;
    forceUpdate();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return [contextMenuCbRef.current, getOpenCallback];
}

/**
 * @param event the event to check if it is a @see {@link React.MouseEvent}.
 * @returns if a @see {@link React.MouseEvent} returns true and type casts; otherwise false.
 */
export const isReactMouseEvent = (
  event: React.MouseEvent | MouseEvent
): event is React.MouseEvent => ((event as unknown) as { nativeEvent })?.nativeEvent !== undefined;

/**
 * Closes any opened context menus.
 */
export const closeContextMenu = (): void => {
  // remove the context menu content from the dom
  const elements = [
    ...document.getElementsByClassName(`${Classes.POPOVER2_CONTENT}`),
    ...document.getElementsByClassName(`${Classes.POPOVER2_BACKDROP}`)
  ];

  elements.forEach((e, idx) => {
    elements[idx].innerHTML = '';
  });

  // remove any context menus that were opened by blueprintjs
  hideContextMenu();
};

const InternalImperativeContextMenu = function InternalImperativeContextMenu<T = undefined>(
  props: ImperativeContextMenuProps<T>
): JSX.Element {
  const { content, getOpenCallback, onClose, onOpen } = props;

  const forceUpdate = useForceUpdate();

  const menuChildRef = React.useRef<HTMLDivElement>();

  const dataRef = React.useRef<T>();

  const open: ImperativeContextMenuOpenFunc<T> = React.useCallback(
    (event: React.MouseEvent | MouseEvent, data: T) => {
      event.preventDefault();
      forceUpdate();
      const actualEvent = isReactMouseEvent(event) ? event.nativeEvent : event;
      dataRef.current = data;
      const newEvent: MouseEvent = new MouseEvent(
        // ! must be of type contextmenu for event to fire properly
        'contextmenu',
        {
          // capture the positioning of the event
          clientX: actualEvent.clientX,
          clientY: actualEvent.clientY,
          // ! ensure that the events bubbles up so that contextmenu receives the event
          bubbles: true
        }
      );
      menuChildRef.current?.dispatchEvent(newEvent);
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  );

  // on mount use effect
  // pass the open function back to the parent prop so that it can be called to open the menu
  React.useEffect(
    () => {
      getOpenCallback(open);
      if (onOpen) {
        onOpen();
      }
      forceUpdate();
    },
    // We only want this to run onMount so we need no dependencies
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  );

  const contentFunc = React.useCallback(() => {
    if (typeof content === 'function') {
      return content(dataRef.current);
    }
    return content;
  }, [content]);

  const handleOnClose = React.useCallback(() => {
    closeContextMenu();
    // TODO: Update this when code to clear brush stroke is refactored
    if (dataRef.current && 'onClose' in (dataRef.current as any)) {
      (dataRef as any).current.onClose();
    }
    if (onClose) {
      onClose();
    }
  }, [onClose]);

  return (
    <ContextMenu2
      content={contentFunc}
      onClose={handleOnClose}
      // !This 'unknown' is needed to trick ContextMenu2 to return focus since they omit it
      popoverProps={{ shouldReturnFocusOnClose: true } as unknown}
    >
      <div ref={menuChildRef} />
    </ContextMenu2>
  );
};

/**
 * Menu item designed to replace the imperative call from Context Menu that was deprecated in ContextMenu2
 */
export const ImperativeContextMenu = React.memo(
  InternalImperativeContextMenu
) as typeof InternalImperativeContextMenu;
