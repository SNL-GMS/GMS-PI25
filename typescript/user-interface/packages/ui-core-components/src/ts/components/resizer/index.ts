export { ResizeContainer } from './resize-container';
export { ResizeContext } from './resize-context';
export { BottomResizeHandle } from './resize-handle';
export { Resizer } from './resizer';
export * from './types';
