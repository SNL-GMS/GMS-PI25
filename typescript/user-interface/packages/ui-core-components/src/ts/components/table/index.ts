export { ClientSideTable as Table } from './client-side-table';
export * from './infinite-table';
export * from './percent-bar';
export * from './table-cell-renderer';
export * from './types';
export * from './use-datasource';
export * as AgGridCommunity from 'ag-grid-community';
export { AgGridReact } from 'ag-grid-react';
