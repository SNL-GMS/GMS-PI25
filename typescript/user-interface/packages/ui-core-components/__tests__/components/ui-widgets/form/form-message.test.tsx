import { Intent } from '@blueprintjs/core';
import { render } from '@testing-library/react';
import React from 'react';

import { FormMessage } from '../../../../src/ts/components';

describe('Form message', () => {
  it('can render with undefined', () => {
    const { container } = render(<FormMessage message={undefined} />);
    expect(container).toMatchSnapshot();
  });

  it('can render with no details and intent', () => {
    const { container } = render(<FormMessage message={{ summary: 'my summary' }} />);
    expect(container).toMatchSnapshot();
  });

  it('can render with no intent', () => {
    const { container } = render(
      <FormMessage message={{ summary: 'my summary', details: 'my details' }} />
    );
    expect(container).toMatchSnapshot();
  });

  it('can render with with different intents', () => {
    let rendered = render(
      <FormMessage
        message={{ summary: 'my summary', details: 'my details', intent: Intent.NONE }}
      />
    );
    expect(rendered.container).toMatchSnapshot();

    rendered = render(
      <FormMessage
        message={{ summary: 'my summary', details: 'my details', intent: Intent.PRIMARY }}
      />
    );
    expect(rendered.container).toMatchSnapshot();

    rendered = render(
      <FormMessage
        message={{ summary: 'my summary', details: 'my details', intent: Intent.SUCCESS }}
      />
    );
    expect(rendered.container).toMatchSnapshot();

    rendered = render(
      <FormMessage
        message={{ summary: 'my summary', details: 'my details', intent: Intent.WARNING }}
      />
    );
    expect(rendered.container).toMatchSnapshot();

    rendered = render(
      <FormMessage
        message={{ summary: 'my summary', details: 'my details', intent: Intent.DANGER }}
      />
    );
    expect(rendered.container).toMatchSnapshot();
  });
});
