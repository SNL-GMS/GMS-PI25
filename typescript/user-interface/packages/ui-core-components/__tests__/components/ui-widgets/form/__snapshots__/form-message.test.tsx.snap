// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form message can render with no details and intent 1`] = `
<div>
  <div
    class="form-message"
  >
    <span
      class="form-message__summary"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-info-sign"
        icon="info-sign"
      >
        <svg
          data-icon="info-sign"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zM7 3h2v2H7V3zm3 10H6v-1h1V7H6V6h3v6h1v1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      my summary
    </span>
  </div>
</div>
`;

exports[`Form message can render with no intent 1`] = `
<div>
  <div
    class="form-message"
  >
    <span
      class="form-message__summary"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-info-sign"
        icon="info-sign"
      >
        <svg
          data-icon="info-sign"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zM7 3h2v2H7V3zm3 10H6v-1h1V7H6V6h3v6h1v1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      my summary
    </span>
    <span
      class="form-message__recommendation"
    >
      my details
    </span>
  </div>
</div>
`;

exports[`Form message can render with undefined 1`] = `<div />`;

exports[`Form message can render with with different intents 1`] = `
<div>
  <div
    class="form-message"
  >
    <span
      class="form-message__summary"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-info-sign"
        icon="info-sign"
      >
        <svg
          data-icon="info-sign"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zM7 3h2v2H7V3zm3 10H6v-1h1V7H6V6h3v6h1v1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      my summary
    </span>
    <span
      class="form-message__recommendation"
    >
      my details
    </span>
  </div>
</div>
`;

exports[`Form message can render with with different intents 2`] = `
<div>
  <div
    class="form-message form-message--info"
  >
    <span
      class="form-message__summary"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-info-sign bp4-intent-primary"
        icon="info-sign"
      >
        <svg
          data-icon="info-sign"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zM7 3h2v2H7V3zm3 10H6v-1h1V7H6V6h3v6h1v1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      my summary
    </span>
    <span
      class="form-message__recommendation"
    >
      my details
    </span>
  </div>
</div>
`;

exports[`Form message can render with with different intents 3`] = `
<div>
  <div
    class="form-message form-message--info"
  >
    <span
      class="form-message__summary"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-info-sign bp4-intent-success"
        icon="info-sign"
      >
        <svg
          data-icon="info-sign"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zM7 3h2v2H7V3zm3 10H6v-1h1V7H6V6h3v6h1v1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      my summary
    </span>
    <span
      class="form-message__recommendation"
    >
      my details
    </span>
  </div>
</div>
`;

exports[`Form message can render with with different intents 4`] = `
<div>
  <div
    class="form-message form-message--warning"
  >
    <span
      class="form-message__summary"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-warning-sign bp4-intent-warning"
        icon="warning-sign"
      >
        <svg
          data-icon="warning-sign"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M15.84 13.5l.01-.01-7-12-.01.01c-.17-.3-.48-.5-.85-.5s-.67.2-.85.5l-.01-.01-7 12 .01.01c-.09.15-.15.31-.15.5 0 .55.45 1 1 1h14c.55 0 1-.45 1-1 0-.19-.06-.35-.15-.5zm-6.85-.51h-2v-2h2v2zm0-3h-2v-5h2v5z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      my summary
    </span>
    <span
      class="form-message__recommendation"
    >
      my details
    </span>
  </div>
</div>
`;

exports[`Form message can render with with different intents 5`] = `
<div>
  <div
    class="form-message form-message--danger"
  >
    <span
      class="form-message__summary"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-error bp4-intent-danger"
        icon="error"
      >
        <svg
          data-icon="error"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M7.99-.01c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm1 13h-2v-2h2v2zm0-3h-2v-7h2v7z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      my summary
    </span>
    <span
      class="form-message__recommendation"
    >
      my details
    </span>
  </div>
</div>
`;
